Lib/ufo2ft/util.py

```diff
@@ -431,3 +433,45 @@ def _kwargsEval(s):
     return eval(
         "dict(%s)" % s, {"__builtins__": {"True": True, "False": False, "dict": dict}}
     )
+
+
+_pluginSpecRE = re.compile(
+    r"(?:([\w\.]+)::)?"  # MODULE_NAME + '::'
+    r"(\w+)"  # CLASS_NAME [required]
+    r"(?:\((.*)\))?"  # (KWARGS)
+)
+
+
+def _loadPluginFromString(spec, moduleName, isValidFunc):
+    spec = spec.strip()
+    m = _pluginSpecRE.match(spec)
+    if not m or (m.end() - m.start()) != len(spec):
+        raise ValueError(spec)
+    moduleName = m.group(1) or moduleName
+    className = m.group(2)
+    kwargs = m.group(3)
+
+    module = importlib.import_module(moduleName)
+    klass = getattr(module, className)
+    if not isValidFunc(klass):
+        raise TypeError(klass)
+    try:
+        options = _kwargsEval(kwargs) if kwargs else {}
+    except SyntaxError:
+        raise ValueError("options have incorrect format: %r" % kwargs)
+
+    if hasattr(klass, "_args"):
+        # Process positional arguments
```

Comment:
this posargs handling should be moved to the base classes constructors, so they can also be initialized using keywords.. But then you'd have to dumplicate in both BaseFeatureWriter and BaseFilter `__init__`.. I guess it's ok like that

Reply:
actually, only the filters can have positional arguments at all (have that `_args` attribute that you're checking in here). Yeah, i think it makes more sense to move this logic to base filter class init. Do you mind doing that?

Reply:
BaseFeatureWriter and BaseFilter should probably inherit from a common BasePlugin class. Should that be in the PR or another?

Reply:
yeah, that'd be nice too. I'd say do it in same PR

Reply:
if you can't reconcile the differences between the two base classes, then just move this pos-to-kwargs logic to the base filter init

Reply:
OK. I removed posargs logic from _loadPluginFromString and changed BaseFilter. That's much nicer.

Reply:
it seems you're only handling the case where all the positionals are specified as keywords, see my other comment https://github.com/googlefonts/ufo2ft/pull/466#commitcomment-47603167